### ISSUE TITLE
upgrade from 3.1 to pre-3.2:: migration_manager err & bad_enum_set_mask in upgrade_test

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#5291]
*Installation details*
Scylla version (or git commit hash):  Upgrade from ` 3.1.1-0.20191024.f32ec885c` to `666.development-0.20191114.552c56633e0`
Cluster size: 4
OS (RHEL/CentOS/Ubuntu/AWS AMI): Ubuntu16.04

#### Ling cluster:
- https://console.cloud.google.com/compute/instances?project=skilled-adapter-452&instancessize=50&instancesquery=%255B%257B_22k_22_3A_22_22_2C_22t_22_3A10_2C_22v_22_3A_22_5C_222e2a90b2_5C_22_22%257D%255D
- Node rolling-upgrade-upgrade--ubuntu-xen-db-node-2e2a90b2-0-1 [35.231.126.16 | 10.142.0.151]
- Node rolling-upgrade-upgrade--ubuntu-xen-db-node-2e2a90b2-0-2 [34.73.61.20 | 10.142.0.152]
- Node rolling-upgrade-upgrade--ubuntu-xen-db-node-2e2a90b2-0-3 [35.231.169.43 | 10.142.0.153]
- Node rolling-upgrade-upgrade--ubuntu-xen-db-node-2e2a90b2-0-4 [34.73.67.180 | 10.142.0.154]
- Node rolling-upgrade-upgrade--ubuntu-xen-loader-node-2e2a90b2-0-1 [34.74.166.172 | 10.142.0.155]
- Node rolling-upgrade-upgrade--ubuntu-xen-monitor-node-2e2a90b2-0-1 [35.237.95.42 | 10.142.0.156]

#### DB ERRORs after upgraded one db node:
- migration_manager - Failed to update definitions from 10.142.0.151:1: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 723420b4-8501-3ecd-92dd-b01de2e227f6): std::out_of_range (regular column id 1 >= 1)
- rpc - client 10.142.0.151:51561 msg_id 85438:   got exception while processing a message: bad_enum_set_mask (Bit mask contains invalid enumeration indices.)

#### Test Scenario:
- execute verify test of https://github.com/scylladb/scylla-cluster-tests/blob/master/sdcm/fill_db_data.py
- upgrade one db node (db-node-2)
- re-execute verify test of https://github.com/scylladb/scylla-cluster-tests/blob/master/sdcm/fill_db_data.py

#### Logs:

```
11-16 00:12:31,618  > Populate DB with many types of tables and data
11-16 00:15:40,592  > Run some Queries to verify data BEFORE UPGRADE
11-16 00:16:43,696  > Re-Populate DB with many types of tables and data

11-16 00:34:05,069  > Upgrade Node rolling-upgrade-upgrade--ubuntu-xen-db-node-2e2a90b2-0-1 begin
11-16 00:34:05,070  > Upgrading a Node
11-16 00:36:47,153  > Upgrade Node rolling-upgrade-upgrade--ubuntu-xen-db-node-2e2a90b2-0-1 ended

11-16 00:55:31,157  > Run some Queries to verify data after upgraded one node

11-16T00:55:34+00:00  rolling-upgrade-upgrade--ubuntu-xen-db-node-2e2a90b2-0-3 !ERR    
 | scylla:  [shard 1] migration_manager - Failed to update definitions from 10.142.0.151:1: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 723420b4-8501-3ecd-92dd-b01de2e227f6): std::out_of_range (regular column id 1 >= 1)

11-16T01:01:42+00:00  rolling-upgrade-upgrade--ubuntu-xen-db-node-2e2a90b2-0-2 !INFO  
  | scylla:  [shard 1] rpc - client 10.142.0.151:51561 msg_id 85438:   got exception while processing a message: bad_enum_set_mask (Bit mask contains invalid enumeration indices.)
....
11-16T01:01:52+00:00  rolling-upgrade-upgrade--ubuntu-xen-db-node-2e2a90b2-0-2 !INFO  
  | scylla:  [shard 1] rpc - client 10.142.0.151:51561 msg_id 859267:  got exception while processing a message: bad_enum_set_mask (Bit mask contains invalid enumeration indices.)
```

#### Test Logs:
- db logs: https://cloudius-jenkins-test.s3.amazonaws.com/2e2a90b2-1b0a-414b-83ac-cec61ad11c20/rolling-upgrade-upgrade--ubuntu-xen-db-cluster-2e2a90b2.zip
- sct event logs: https://cloudius-jenkins-test.s3.amazonaws.com/2e2a90b2-1b0a-414b-83ac-cec61ad11c20/events_log.zip
- sct job log: https://cloudius-jenkins-test.s3.amazonaws.com/2e2a90b2-1b0a-414b-83ac-cec61ad11c20/job_log.zip      

@roydahan @slivne

**Comment by amoskong** [comment]
The two kinds of db logs are all related with the upgraded db node (10.142.0.151).

**Comment by amoskong** [comment]
This issue is reproduced in https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-ubuntu18.04/5/
- upgrade from 3.0.10-0.20190815.b3bfd8c08 to 3.2.rc1-0.20191124.85b1a45252

```
11-25 08:24:17 > Populate DB with many types of tables and data
11-25 08:27:50 > Run some Queries to verify data BEFORE UPGRADE
11-25 08:28:50 > Re-Populate DB with many types of tables and data
11-25 08:46:35 > Upgrade Node rolling-upgrade-upgrade--ubuntu-bio-db-node-15cf97fd-0-1 begin
11-25 08:46:44 > Upgrading a Node
11-25 08:49:34 > Upgrade Node rolling-upgrade-upgrade--ubuntu-bio-db-node-15cf97fd-0-1 ended
11-25 09:07:31 > Run some Queries to verify data after upgraded one node

11-25 09:13:04 > 2019-11-25T09:13:03+00:00  rolling-upgrade-upgrade--ubuntu-bio-db-node-15cf97fd-0-4 !WARNING | scylla:  [shard 1]
 seastar - Exceptional future ignored: bad_enum_set_mask (Bit mask contains invalid enumeration indices.), 
backtrace:   0x59d09e#012  0x37c218#012  0x307709e#012  0x37aaee#012  0x37acb1#012  0x44b20f#012  0x46e66a#012  0x52790d#012  /lib/x86_64-linux-gnu/libpthread.so.0+0x76da#012  /lib/x86_64-linux-gnu/libc.so.6+0x12188e
```

**Comment by slivne** [comment]
@eliransin is this the same as https://github.com/scylladb/scylla/issues/5288

**Comment by bhalevy** [comment]
@tgrabiec can the following explained by the new cdc column (#5288)?
```
migration_manager - Failed to update definitions from 10.142.0.151:1: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 723420b4-8501-3ecd-92dd-b01de2e227f6): std::out_of_range (regular column id 1 >= 1)
```

**Comment by tgrabiec** [comment]
@bhalevy Yes

**Comment by avikivity** [comment]
Fixed by f0d8dd409405a637e27d3092b9d71e2461d53d33

**Comment by avikivity** [comment]
3.2 backport in edf431f5810762397a02aff9b5eb0892831e814f
